vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSInstallState.java

```diff
@@ -417,4 +422,84 @@ public WorkspaceFilter getFilter() {
     public Properties getProperties() {
         return properties;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
```

Comment:
Just for fun: I always thought using 31 as a factor is a quite bad choice, as it easily creates conflicts. https://stackoverflow.com/a/2816747/21499 explains why I think 92821 is much better when you are using that pattern. :-)